haystack/nodes/answer_generator/openai.py

```diff
@@ -137,7 +142,18 @@ def predict(self, query: str, documents: List[Document], top_k: Optional[int] =
         }
 
         headers = {"Authorization": f"Bearer {self.api_key}", "Content-Type": "application/json"}
-        response = requests.request("POST", url, headers=headers, data=json.dumps(payload))
+        try:
+            response = requests.request("POST", url, headers=headers, data=json.dumps(payload))
+            time.sleep(60 / self.max_calls_per_minute)
+        except Exception as e:
```

Comment:
Can we catch the exception type / API response code for rate-limits? If yes, we can make the error message here more actionable like "Seems like the OpenAI API rate-limited your requests. See limits of plans here (link) and consider setting `max_calls_per_minute` to a lower value (currently: XX) when initializing the OpenAIAnswerGenerator node in haystack`

Reply:
Sorry, this try catch is was not meant for catching the rate limit. 
But I see the value of catching the rate limit as you suggest.

I think there is the need to catch more general problems, since when I was running eval after adjusting the rate limit, the API also ran into errors from time to time. That's why here is a very general try-catch block.